### PR TITLE
move EFI random seed update logic from systemd-random-seed.service into separate service

### LINKED ISSUE #25769: efi random seed code should be split out of systemd-random-seed.service
### systemd version the issue has been seen with

main

### Used distribution

fedora

### Linux kernel version used

_No response_

### CPU architectures issue was seen on

None

### Component

_No response_

### Expected behaviour you didn't see

Since f913c784ad4c93894fd6cb2590738113dff5a694 the EFI random seed is updated whenever the /var/ seed is updated too.

This should be split into a separate service. systemd-random-seed.service runs very early at boot, before the ESP or the autofs for it are mounted, hence it simply won't work right now. We could certainly add an ordering dep towards the automount, but that would mean we'd delay the regular randoms seed handling after the ESP has shown up, i.e. even later. Moreover the ESP mount point is not clearly defined, hence it's hard to express a clear dependency.

Hence, easiest is to just turn the EFI random seed code into its own service that is run after local-fs.target. This also allows us to run it during the regular boot phase (which is OK, the EFI random seed is not time critical after all)

### Unexpected behaviour you saw

_No response_

### Steps to reproduce the problem

_No response_

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### PR DESCRIPTION
**Pull request by poettering**
This fixes #25769, and moves updating of the EFI random seed out of systemd-random-seed.service into systemd-boot-random-seed.service which just calls "bootctl random-seed" to reresh the seed.

This is done so that systemd-random-seed.service can run as early as possible, only requiring writable /var/ and nothing else. OTOH systemd-boot-random-seed.service can run much later, after the ESP has been set up via autofs and everything.

This is mostly implemented by renaming "systemd-boot-system-token.service" to "systemd-boot-random-seed.service", and making minor changes to its conditionioning rules. Previously the service's focus was to set up the system token EFI var (and only as side-effect also set up the EFI random seed file). We swap this around now to have the focus on writing the random seed file (and only as side-effect also set up the system token EFI var if missing). It's now started on every boot, as opposed to only on first boot.

Fixes: #25769

### CODE REVIEW

**Comment by poettering** [comment]
This primarily moves around the code added in f913c784ad4c93894fd6cb2590738113dff5a694

File: src/boot/bootctl-random-seed.c [hunk 1]
```diff
@@ -117,16 +75,106 @@
...
+
+        if (syncfs(fd) < 0)
+                return log_error_errno(errno, "Failed to sync ESP file system: %m");
+
+        log_info("Random seed file %s/loader/random-seed successfully %s (%zu bytes).", esp, refreshed ? "refreshed" : "written", sizeof(buffer));
+
+        return set_system_token();
+
+fail:
+        if (loader_dir_fd >= 0 && tmp)
```
[the remark this answers is not in the document]
Comment (bluca) [reply src/boot/bootctl-random-seed.c]: unaddressed

**Review by bluca — APPROVED** [review]

**Comment by poettering** [comment]
addressed now. thanks!

**Comment by zx2c4** [comment]
Sorry I didn't look at this earlier, but avoiding duplication in that functionality seems like a good idea. I would have liked to see it all moved into a single service, rather than having to have two, but it looks like there are some trixy partition readiness conditions that makes doing two services a lot easier. I also like the `sha256_process_bytes_and_size` helper introduced.

**Pull request merged**